src/filters/common.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import HtmlDataProcessor from '@ckeditor/ckeditor5-engine/src/dataprocessor/htmldataprocessor';
+
+const htmlDataProcessor = new HtmlDataProcessor();
+
+/**
+ * Extracts `body` tag contents from the provided HTML string.
+ *
+ * @param {Object} data
+ * @param {String} data.html HTML string from which `body` contents will be extracted.
+ * @returns {Object} result
+ * @returns {String|null} result.body Extracted `body` contents. If `body` tag was not present or empty, `null` is returned.
+ */
+export function extractBody( data ) {
```

Comment:
I wonder whether this function could/should use `DOMParser`.  If so, it'd be good to use `DOMParser` once for the whole process, so it'd have to look a bit differently.

Reply:
It should make sense, but then it will have to return something like `DocumentFragment` or a `View` instance. So it will be `extract+parseBody` basically. It seems for now that there is no need to operate on `body` string so we may just combine these two steps into one (because `HtmlDataProcessor` uses `DOMParser` internally). WDYT?